blocks/card-grid/card-grid.js

```diff
@@ -0,0 +1,109 @@
+import {
+  loadBlock, toClassName,
+} from '../../scripts/lib-franklin.js';
+import ffetch from '../../scripts/ffetch.js';
+import { createCardBlock } from '../card/card.js';
+
+export default async function decorate(block) {
+  // apply modifiers to the wrapper as well
+  const gapCls = [...block.classList].filter((cls) => cls.indexOf('gap') >= 0);
+  if (gapCls.length) block.parentElement.classList.add(gapCls);
+  // Formats a table and applies background images. Generally, each item should be in a table cell.
+  // When a cell is empty, the cells are merged vertically.
+  const columns = [...block.firstElementChild.children];
+
+  // cells in each column are moved into the first row as <li>
+  block.querySelectorAll(':scope > div').forEach((row, rowIndex) => {
+    row.querySelectorAll(':scope > div').forEach((cell, columnIndex) => {
+      wrapContentInList(cell);
+
+      // make sure it's not rendered as a button
+      cell.querySelector('a')?.classList.remove('button');
+
+      if (rowIndex > 0) {
+        // move to first row
+        cell.querySelectorAll('li').forEach((li) => {
+          if (li.childElementCount) {
+            columns[columnIndex].querySelector('ul').append(li);
+          }
+        });
+        cell.remove();
+      }
+    });
+    if (rowIndex > 0) {
+      row.remove();
+    }
+  });
+  removeEmptyLi(block);
```

Comment:
Please don't put lists in table cells. Instead use on cell for each link. Then you can also remove the ugly <li> handling in this code.

Reply:
removed everything and rebuild with minimal nodes.

Reply:
fixed